contracts/utils/NameEncoder.sol

```diff
@@ -0,0 +1,104 @@
+pragma solidity ^0.8.4;
+
+import "./Strings.sol";
+
+library NameEncoder {
+    using strings for *;
+
+    function encodePart(strings.slice memory name)
```

Comment:
This is clever, but does O(n^2) memory allocation. Which is okay for read ops, but wouldn't be great onchain.